packages/odyssey-react-mui/src/ThemeProvider.tsx

```diff
@@ -10,11 +10,34 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
+import createCache from "@emotion/cache";
+import { CacheProvider } from "@emotion/react";
 import { ThemeProvider as MuiThemeProvider } from "@mui/material/styles";
-import { ReactElement } from "react";
+import { memo, ReactElement, useMemo } from "react";
 
 import { odysseyTheme } from "./theme";
+import { useUniqueAlphabeticalId } from "./useUniqueAlphabeticalId";
 
-export const ThemeProvider = ({ children }: { children: ReactElement }) => (
-  <MuiThemeProvider theme={odysseyTheme}>{children}</MuiThemeProvider>
-);
+const ThemeProvider = ({ children }: { children: ReactElement }) => {
+  const uniqueAlphabeticalId = useUniqueAlphabeticalId();
+
+  const emotionCache = useMemo(
+    () =>
+      createCache({
+        key: uniqueAlphabeticalId,
+        // @ts-expect-error ts(2345)
+        nonce: window.cspNonce,
+      }),
+    [uniqueAlphabeticalId]
+  );
+
+  return (
+    <CacheProvider value={emotionCache}>
```

Comment:
what about also allowing the emotionCache to also be passed in?

Reply:
I thought about that, by why would we want people passing in their own cache? The purpose of wrapping `ThemeProvider` is to do all the heavily-lifting ourselves for ease-of-development.

I might separate out this into a separate `CacheProvider` component though. While it's more trouble, it isn't tightly-coupled and will make the Storybook hack less of an issue.

Reply:
i just had to pass in the cache to our `CacheProvider`.
The classes created by `@emotion/css` don't use the `CacheProvider` cache. I had to use the cache returned from `@emotion/css/create-instance` and pass that to `CacheProvider` 
but now in hindsight, maybe I should have used the `css` returned from `@emotion/react`. 😢